src/smrt/shift_driver.cpp

```diff
@@ -93,17 +79,46 @@ void ShiftDriver::solve(const std::vector<double>& th_temperature,
       double total_power = std::accumulate(mean.begin(), mean.end(), 0.0);
 
       for (int cellid = 0; cellid < mean.size(); ++cellid) {
-        Expects(cellid < d_power_map.size());
-        const auto& elem_list = d_power_map[cellid];
         double tally_volume = d_geometry->cell_volume(cellid);
-
-        for (const auto& elem : elem_list) {
-          Expects(elem < power.size());
-          power[elem] = mean[cellid] / tally_volume / total_power;
-        }
+        d_power_by_cell_ID.push_back(mean[cellid] / tally_volume / total_power);
       }
     }
   }
+
+  normalize_heat_source(d_power_by_cell_ID, power);
```

Comment:
I actually think it would be better to get rid of `normalize_heat_source` entirely - I introduced it a bit back because I thought it might help with the fact that Shift's power normalization originally needed information from Nek. The latest commits combine `normalize_heat_source` and `heat_source`

Reply:
Ah yes, I was also thinking that :+1: